### PR TITLE
Refactor Transform.compose to have more specific typing

### PR DESCRIPTION
**Pull request by FayCarsons**
Refactored `compose` type signature from generic fn composition to shape-specific fn composition.

### CODE REVIEW

File: lib/transform.ml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 open Shape
 
+type transform = (shape -> shape)
```
Comment (nikochiko) [inline lib/transform.ml:3]: This is already in the mli file. Does it need to be here?

File: lib/joy.mli [hunk 2]
```diff
@@ -12,7 +12,7 @@ val complex : shapes -> shape
 val rotate : int -> shape -> shape
 val translate : float -> float -> shape -> shape
 val scale : float -> shape -> shape
-val compose : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
+val compose : (shape -> shape) -> (shape -> shape) -> shape -> shape
```
Comment (nikochiko) [inline lib/joy.mli:15]: Can we type this as `transformation -> transformation -> trasnformation` ?

File: lib/transform.mli [hunk 3]
```diff
@@ -1,5 +1,6 @@
-val translate : float -> float -> Shape.shape -> Shape.shape
-val scale : float -> Shape.shape -> Shape.shape
-val rotate : int -> Shape.shape -> Shape.shape
-val compose : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
-val repeat : int -> (Shape.shape -> Shape.shape) -> Shape.shape -> Shape.shape
+type transform = (Shape.shape -> Shape.shape)
```
Comment (nikochiko) [inline lib/transform.mli:1]: I like `transformation` better for the name of a type. `transform` would have been appropriate for the name of a function.

**Review by nikochiko — CHANGES_REQUESTED** [review]

File: lib/transform.ml (hunk 1 above)
Comment (FayCarsons) [reply lib/transform.ml:3]: I don't know, can a type just be declared in the .mli file and not in the .ml file? I had assumed it had to be present in both.

Comment (FayCarsons) [reply lib/transform.ml:3]: It doesn't compile unless its provided in both.

**Comment by FayCarsons** [comment]
Applied suggestions from review, should be good to go!

**Comment by nikochiko** [comment]
Thanks! @FayCarsons

**Pull request merged**